django/test/runner.py

```diff
@@ -336,12 +336,20 @@ def run(self, test):
         return result
 
 
-def parallel_type(value):
-    """Parse value passed to the --parallel option."""
+def get_num_test_processes(value=None):
+    """Number of test processes when using the --parallel option."""
     # The current implementation of the parallel test runner requires
     # multiprocessing to start subprocesses with fork().
     if multiprocessing.get_start_method() != 'fork':
         return 1
+    try:
+        return int(os.environ['DJANGO_TEST_PROCESSES'])
+    except KeyError:
+        return value if value else multiprocessing.cpu_count()
+
+
+def parallel_type(value):
+    """Parse value passed to the --parallel option."""
     if value == 'auto':
         return multiprocessing.cpu_count()
```

Comment:
Is it too late to move the conversion of `auto` to an integer to a post-processing step (e.g. your `get_num_test_processes()` function)? I feel like the `parallel_type` function's job here should only be to check that the value equals `auto` if the value is a string, but not to apply the environment-dependent business logic to convert `auto` to a number. (I also see that `get_num_test_processes()` is already calling `multiprocessing.cpu_count()`, so there may be some duplication of logic with the way things are currently structured.)

Reply:
Let's try. I pushed edits.